Subject: patches/sched-rt-stats.patch


So, I have merged my previous patch (to display rt_nr_running info in
sched_debug.c) with this one.

Signed-off-by: Ankita Garg <ankita@in.ibm.com>
[mingo@elte.hu: fix it to work on !SCHEDSTATS too]
Signed-off-by: Ingo Molnar <mingo@elte.hu>
--
 kernel/sched_debug.c |   13 +++++++++++++
 1 file changed, 13 insertions(+)

Index: linux-2.6-tip/kernel/sched_debug.c
===================================================================
--- linux-2.6-tip.orig/kernel/sched_debug.c
+++ linux-2.6-tip/kernel/sched_debug.c
@@ -280,6 +280,19 @@ static void print_cpu(struct seq_file *m
 	P(cpu_load[2]);
 	P(cpu_load[3]);
 	P(cpu_load[4]);
+#ifdef CONFIG_PREEMPT_RT
+	/* Print rt related rq stats */
+	P(rt.rt_nr_running);
+	P(rt.rt_nr_uninterruptible);
+# ifdef CONFIG_SCHEDSTATS
+	P(rto_schedule);
+	P(rto_schedule_tail);
+	P(rto_wakeup);
+	P(rto_pulled);
+	P(rto_pushed);
+# endif
+#endif
+
 #undef P
 #undef PN
 
